src/test/java/pages/AbstractPage.java

```diff
@@ -0,0 +1,30 @@
+package pages;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public abstract class AbstractPage {
+    public static final long WAIT_TIMEOUT_SECONDS = 50;
+    protected WebDriver driver;
+
+    protected abstract AbstractPage openPage();
+
+    public void waitForPageLoad(WebDriver driver) {
+
+        ExpectedCondition<Boolean> pageLoadCondition = new ExpectedCondition<Boolean>() {
+            public Boolean apply(WebDriver driver) {
+                return ((JavascriptExecutor) driver).executeScript("return document.readyState").equals("complete");
+            }
+        };
+        WebDriverWait wait = new WebDriverWait(driver, WAIT_TIMEOUT_SECONDS);
+        wait.until(pageLoadCondition);
+    }
+
+    protected AbstractPage(WebDriver driver) {
```

Comment:
move this constructor to line 12+, wright after variables declaration

Reply:
Refactored, thank you.
Please see in the following commit.